src/plugins/clearURLs/index.ts

```diff
@@ -51,42 +63,25 @@ export default definePlugin({
             : (str || "");
     },
 
-    createRules() {
-        // Can be extended upon once user configs are available
-        // Eg. (useDefaultRules: boolean, customRules: Array[string])
-        const rules = defaultRules;
-
-        this.universalRules = new Set();
-        this.rulesByHost = new Map();
-        this.hostRules = new Map();
-
-        for (const rule of rules) {
-            const splitRule = rule.split("@");
-            const paramRule = new RegExp(
-                "^" +
-                this.escapeRegExp(splitRule[0]).replace(/\\\*/, ".+?") +
-                "$"
-            );
+    async createRules() {
+        const res = await fetch(url).then(res => res.json()) as { providers: Providers; };
 
-            if (!splitRule[1]) {
-                this.universalRules.add(paramRule);
-                continue;
-            }
-            const hostRule = new RegExp(
-                "^(www\\.)?" +
-                this.escapeRegExp(splitRule[1])
-                    .replace(/\\\./, "\\.")
-                    .replace(/^\\\*\\\./, "(.+?\\.)?")
-                    .replace(/\\\*/, ".+?") +
-                "$"
-            );
-            const hostRuleIndex = hostRule.toString();
+        this.providers = [];
+
+        for (const [name, provider] of Object.entries(res.providers) as [string, Provider][]) {
```

Comment:
this cast shouldn't be necessary
```suggestion
        for (const [name, provider] of Object.entries(res.providers)) {
```